frontend/src/widgets/resultaggregator.js

```diff
@@ -22,176 +22,164 @@ import WidgetHeader from '../components/widget-header';
 import ParamDropdown from '../components/param-dropdown';
 
 
-export class ResultAggregatorWidget extends React.Component {
-  static propTypes = {
-    title: PropTypes.string,
-    params: PropTypes.object,
-    dropdownItems: PropTypes.array,
-    onDeleteClick: PropTypes.func,
-    onEditClick: PropTypes.func
-  };
+const ResultAggregatorWidget = ( props ) => {
+  const {
+    title,
+    params,
+    dropdownItems,
+    onDeleteClick,
+    onEditClick
+  } = props;
+
 
-  constructor (props) {
-    super(props);
-    this.title = props.title || 'Result Categories';
-    this.params = props.params || {};
-    this.getResultData = this.getResultData.bind(this);
-    this.onDaySelect = this.onDaySelect.bind(this);
-    this.onGroupFieldSelect = this.onGroupFieldSelect.bind(this);
-    this.state = {
-      data: {},
-      chartData: [],
-      legendData: [],
-      isLoading: true,
-      total: 0,
-    };
-  }
+  const [chartData, setChartData] = useState([]);
+  const [legendData, setLegendData] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [total, setTotal] = useState(0);
+  const [resultAggregatorError, setResultAggregatorError] = useState(false);
+  const [days, setDays] = useState(params.days);
+  const [groupField, setGroupField] = useState(params.group_field);
 
-  getResultData () {
-    this.setState({isLoading: true});
-    HttpClient.get([Settings.serverUrl, 'widget', 'result-aggregator'], this.params)
+  useEffect(() => {
+    setIsLoading(true);
+    HttpClient.get([Settings.serverUrl, 'widget', 'result-aggregator'], params)
```

Comment:
Compose the object for query string

`{chart_type: chart_type, group_field: group_field, days: days}`

To exclude params from dependencies list - but you'll have to spread any of the other fields into state too.